### PR TITLE
Update pre commit

### PR DESCRIPTION
**Pull request by colin99d**
# Description

This PR will allow us to check all config_terminal settings before a push. This is also for visibility so that we can determine if there is a better way to do this.

# How has this been tested?

Experimented with incorrect settings in the check.

# Checklist:

- [x] Update [our Hugo documentation](https://gamestonkterminal.github.io/GamestonkTerminal/) following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/website).
- [x] Update our tests following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [x] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

File: gamestonk_terminal/config_terminal.py [hunk 1]
```diff
@@ -78,9 +78,8 @@
 DG_TOTP_SECRET = os.getenv("GT_DG_TOTP_SECRET") or None
 
 # https://developer.oanda.com
-OANDA_ACCOUNT_TYPE = (
-    os.getenv("GT_OANDA_ACCOUNT_TYPE") or "practice"
-)  # "live" or "practice"
+DEBUG_MODE = False
```
Comment (DidierRLopes) [inline gamestonk_terminal/config_terminal.py:81]: I think this was not intentional

**Comment by jmaslek** [comment]
Just had a thought.  Instead of having this debug as a feature flag, can we just have a --debug option on the terminal.py ?

**Comment by colin99d** [comment]
> Just had a thought. Instead of having this debug as a feature flag, can we just have a --debug option on the terminal.py ? So that I would run `python terminal.py --debug` without worrying about setting this.

This seems like the best option. How would I implement this?

**Comment by jmaslek** [comment]
> This seems like the best option. How would I implement this?

Not 100% sure, but in terminal.py there is the `__main__`  routine.  You probably want to check if --debug is in that and then set some kind of variable to inform that debug should be true.

File: custom_pre_commit/check_config_terminal.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        The setting to check
+    value : str
+        The correct value of the setting
+
+    Returns
+    ----------
+    correct : bool
+        Returns whether the setting was already correct
+    """
+    print(setting)
```
Comment (DidierRLopes) [inline custom_pre_commit/check_config_terminal.py:108]: I don't think you want this

File: custom_pre_commit/check_config_terminal.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+def main():
+    base = os.path.dirname(os.path.dirname(__file__))
+    path = os.path.join(base, "gamestonk_terminal", "config_terminal.py")
+    with open(path) as file:
+        lines = file.readlines()
+        lines = [line.rstrip() for line in lines]
+
+    returns = [check_setting(lines, k, v) for k, v in settings.items()]
+
+    print(returns)
```
Comment (DidierRLopes) [inline custom_pre_commit/check_config_terminal.py:127]: neither this

File: gamestonk_terminal/config_terminal.py [hunk 4]
```diff
@@ -7,10 +7,10 @@
     load_dotenv(env_files[0])
 
 # Set to True to see full stack traces for debugging/error reporting
-DEBUG_MODE = False
+DEBUG_MODE = True
```
Comment (DidierRLopes) [inline gamestonk_terminal/config_terminal.py:10]: False

File: gamestonk_terminal/config_terminal.py [hunk 5]
```diff
@@ -7,10 +7,10 @@
...
 
 # By default the jupyter notebook will be run on port 8888
-PAPERMILL_NOTEBOOK_REPORT_PORT = "8888"
+PAPERMILL_NOTEBOOK_REPORT_PORT = "7757"
```
Comment (DidierRLopes) [inline gamestonk_terminal/config_terminal.py:13]: 8888

File: gamestonk_terminal/config_terminal.py [hunk 6]
```diff
@@ -78,9 +78,8 @@
 DG_TOTP_SECRET = os.getenv("GT_DG_TOTP_SECRET") or None
 
 # https://developer.oanda.com
-OANDA_ACCOUNT_TYPE = (
-    os.getenv("GT_OANDA_ACCOUNT_TYPE") or "practice"
-)  # "live" or "practice"
+OANDA_ACCOUNT_TYPE = os.getenv("GT_OANDA_ACCOUNT") or "REPLACE_ME"
```
Comment (DidierRLopes) [inline gamestonk_terminal/config_terminal.py:81]: GT_OANDA_ACCOUNT_TYPE

File: terminal.py [hunk 7]
```diff
@@ -316,7 +318,12 @@ def terminal(jobs_cmds: List[str] = None):
 
 if __name__ == "__main__":
     if len(sys.argv) > 1:
-        if ".gst" in sys.argv[1]:
+        if "--debug" in sys.argv:
+            os.environ["DEBUG_MODE"] = "true"
+            sys.argv.remove("--debug")
+        else:
+            os.environ["DEBUG_MODE"] = "false"
+        if len(sys.argv) > 1 and ".gst" in sys.argv[1]:
```
Comment (DidierRLopes) [inline terminal.py:326]: Can you remove that first condition, it is already done in line 320, we just need the 2nd part

**Review by DidierRLopes — COMMENTED** [review]
Some comments, but if it works it looks good to me

File: terminal.py (hunk 7 above)
Comment (colin99d) [reply terminal.py:326]: You need it twice because if you remove "--debug" you may be left with only one argument.

Comment (DidierRLopes) [reply terminal.py:326]: Gotchaaa! Sry, missed that.

**Review by DidierRLopes — APPROVED** [review]

**Pull request merged**